plugins/git/gitmessenger.nix

```diff
@@ -0,0 +1,117 @@
+    noDefaultMappings =
+      helpers.defaultNullOpts.mkBool false
+        "When this value is set, it does not define any key mappings";
+    intoPopupAfterShow = helpers.defaultNullOpts.mkBool true ''
+      When this value is set to v:false, running :GitMessenger or <plug>(git-messenger) again after
+      showing a popup does not move the cursor in the window.
+    '';
+    alwaysIntoPopup = helpers.defaultNullOpts.mkBool false ''
+      When this value is set to v:true, the cursor goes into a popup window when running
+      :GitMessenger or <Plug>(git-messenger).
+    '';
+    extraBlameArgs = helpers.defaultNullOpts.mkStr "" ''
+      When this variable is set the contents will be appended to the git blame command. Use it to
+      add options (like -w).
+    '';
+    previewMods = helpers.defaultNullOpts.mkStr "" ''
+      This variable is effective only when opening preview window (on Neovim (0.3.0 or earlier)
+      or Vim).
+
+      Command modifiers for opening preview window. The value will be passed as prefix of :pedit
+      command. For example, setting "botright" to the variable opens a preview window at bottom of
+      the current window. Please see :help <mods> for more details.
+    '';
+
+    maxPopupHeight = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = ''
+        Max lines of popup window in an integer value. Setting null means no limit.
+      '';
+    };
+
+    maxPopupWidth = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = ''
+        Max characters of popup window in an integer value. Setting null means no limit.
+      '';
+    };
+
+    dateFormat = helpers.defaultNullOpts.mkStr "%c" ''
+      String value to format dates in popup window. Please see :help strftime() to know the details
+      of the format.
+    '';
+
+    concealWordDiffMarker = helpers.defaultNullOpts.mkBool true ''
+      When this value is set to v:true, markers for word diffs like [-, -], {+, +} are concealed.
+      Set false when you don't want to hide them.
+
+      Note: Word diff is enabled by typing "r" in a popup window.
+    '';
+
+    floatingWinOps = helpers.defaultNullOpts.mkNullable (types.attrsOf types.anything) "{}" ''
+      Options passed to nvim_open_win() on opening a popup window. This is useful when you want to
+      override some window options.
+    '';
+
+    popupContentMargins = helpers.defaultNullOpts.mkBool true ''
+      Setting true means adding margins in popup window. Blank lines at the top and bottom of popup
+      content are inserted. And every line is indented with one whitespace character. Setting false
+      to this variable removes all the margins.
+    '';
+  };
+
+  config =
+    let
+      cfg = config.plugins.gitmessenger;
+    in
+    mkIf cfg.enable {
+      extraPlugins = [ cfg.package ];
+      globals = {
+        git_messenger_close_on_cursor_moved = helpers.mkIfNonNull cfg.closeOnCursorMoved;
```

Comment:
There's no real need to use the `mkIfNonNull` here. When creating a global, it uses `toLuaObject` under the hood, so if it's null it's just going to not be set.

Reply:
I never really know when to know mkIf vs if (as is the case here, because toLuaObject would use an 'if')...

Reply:
AFAIK `mkIf` makes it not set, where as `if` just sets it to null. However, in this case, it's irrelevant, since `toLuaObject` ignores anything that's set to null (since in lua, setting something to nil is the same as not setting it)